### PR TITLE
FP8 Training Support

### PR DESCRIPTION
**Pull request by OpenMOSE**
Gooday :)

I have implemented FP8 support for RWKV-PEFT.

In LoRA and PISSA, fp8matmul is used for Forward operations.
I hope this can be used as a reference.

There is no need to approve the pull request.

It requires Torch2.4+, Pytorch-lightning=2.4.0, and triton=3.0.0,
which are not recommended environment settings.

In x060 7B @ RTX4090
ctx1024, bsz=4

PISSA
NF4 : 2.02kt/s
INT8: 1.84kt/s
FP8 : 2.50kt/s

LoRA
NF4 : 2.01kt/s
INT8: 1.85kt/s
FP8 : 2.49kt/s

It seems to be 35% faster than Bitsandbytes
Loss is roughly the same as Int8

### CODE REVIEW

**Comment by JL-er** [comment]
Thank u, I will test it as soon as possible.
